File: 백준/Silver/1912. 연속합/연속합.py
import sys



def solution():
  n = int(sys.stdin.readline())
  inputs = list(map(int, sys.stdin.readline().split()))


  dp = [0 for _ in range(n)]
  dp[0] = inputs[0]
  for i in range(1, n):
    dp[i] = max(dp[i - 1] + inputs[i], inputs[i])

  print(max(dp))

solution()

File: 백준/Silver/1912. 연속합/연속합.js
let fs = require('fs');
let inputs = fs.readFileSync('/dev/stdin').toString().split('\n');
const n = Number(inputs[0]);
const numbers = inputs[1].split(' ').map(Number);
const dp = [];

for (let i = 0; i < n; i++) {
    if (i === 0) {
        dp.push(numbers[i]);
    } else {
        dp.push(Math.max(dp[i - 1] + numbers[i], numbers[i]));
    }
}

console.log(Math.max(...dp));